libcontainer/process_linux.go

```diff
@@ -772,7 +771,7 @@ func (p *initProcess) start() (retErr error) {
 	if err := p.comm.syncSockParent.Shutdown(unix.SHUT_WR); err != nil && ierr == nil {
 		return err
 	}
-	if !seenProcReady && ierr == nil {
+	if !p.comm.syncSockParent.procReady && ierr == nil {
```

Comment:
Replacing a simple boolean with something that modifies a struct's internal fields makes this logic seem more complicated (the only thing that reads this bool is this code -- why make it seem like other things might read it?).